crates/matrix-sdk-base/src/rooms/mod.rs

```diff
@@ -569,4 +603,37 @@ mod tests {
         base_room_info.handle_notable_tags(&tags);
         assert!(base_room_info.notable_tags.contains(RoomNotableTags::LOW_PRIORITY).not());
     }
+
+    #[test]
+    fn test_get_user_id_for_state_key() {
+        assert_matches!(get_user_id_for_state_key(""), None);
+        assert_matches!(get_user_id_for_state_key("abc"), None);
+        assert_matches!(get_user_id_for_state_key("@nocolon"), None);
+        assert_matches!(get_user_id_for_state_key("@noserverpart:"), None);
+        assert_matches!(get_user_id_for_state_key("@noserverpart:_suffix"), None);
+
+        let user_id = user_id!("@username:example.org");
+
+        assert_matches!(
+            get_user_id_for_state_key(user_id.as_str()),
+            Some(captured_user_id) => assert_eq!(captured_user_id, user_id));
```

Comment:
Does this not work? (and a few times below)

```suggestion
        assert_eq!(
            get_user_id_for_state_key(user_id.as_str()),
            Some(user_id)
        );
```

Reply:
This required `.as_deref()` to the LHS, because `get_user_id_for_state_key` returns an `OwnedUserId` while the `user_id` was a `&UserId`.